examples/rifle_sxiv.sh

```diff
@@ -31,27 +31,27 @@
 TMPDIR="${TMPDIR:-/tmp}"
 tmp="$TMPDIR/sxiv_rifle_$$"
 
-listfiles () {
-    find -L "///${1%/*}" \( ! -path "///${1%/*}" -prune \) -type f -print |
-      grep -iE '\.(jpe?g|png|gif|webp|tiff|bmp)$' |
-      sort | tee "$tmp"
+is_img_extension () {
+    grep -iE '\.(jpe?g|png|gif|svg|webp|tiff|heif|avif|ico|bmp)$'
 }
 
-is_img () {
-    case "${1##*.}" in
-        "jpg"|"jpeg"|"png"|"gif"|"webp"|"tiff"|"bmp") return 0 ;;
-        *) return 1 ;;
-    esac
+listfiles () {
+    find -L "///${1%/*}" \( ! -path "///${1%/*}" -prune \) -type f -print |
+      is_img_extension | sort | tee "$tmp"
 }
 
 open_img () {
-    is_img "$1" || exit 1
-    trap 'rm -f $tmp' EXIT
-    count="$(listfiles "$1" | grep -nF "$1")"
+    # only go through listfiles() if the file has a valid img extension
+    if echo "$1" | is_img_extension >/dev/null 2>&1; then
```

Comment:
Previously the behavior was that if we decided something's not an image, we exit. This changes that to rely on `count` not being set, right? That seems a bit too implicit to me and `rifle` relies on exit codes to disambiguate what to do with the output. Are you sure we're still getting the exit code `rifle` expects and an explicit `exit` wouldn't be more understandable?

Reply:
> Previously the behavior was that if we decided something's not an image, we exit.

That's a behavior this PR is trying to correct. Sxiv is capable of opening directories, but the script would reject it due to extension check.

> `rifle` relies on exit codes to disambiguate what to do with the output. Are you sure we're still getting the exit code `rifle` expects and an explicit `exit` wouldn't be more understandable?

If `sxiv` fails to open the argument, it will exit with 1. I use this script as a standalone wrapper but I don't use ranger (and thus not rifle either).

However, if the concern is around exit code, the only difference this PR makes is that:

0. Previously if the extension wasn't "valid" we rejected right away and exited with `1`. This meant that certain valid inputs (image without extension/directory etc) would get rejected due to false positive on the extension check.
0. Now we let `sxiv` try to open it, if it doesn't succeed it will exit with `1`.

```console
# valid directory containing images

# before
$ rifle_sxiv ~/pic # rifle script rejects the input
$ echo $?
1

# after
$ rifle_sxiv ~/pic # sxiv opens the directory

# invalid input

# before
$ rifle_sxiv /tmp
$ echo $?
1

# after
$ rifle_sxiv /tmp
sxiv: no more files to display, aborting
$ echo $?
1
```

In other words, the exit code will still be `1` upon failure. Difference being that we let `sxiv` try to open it first instead of deciding to exit ourselves (which leads to some false positives).